core/test_network/src/unique_port.rs

```diff
@@ -0,0 +1,65 @@
+#![allow(clippy::restriction)]
+
+use once_cell::sync::Lazy;
+use std::net::{Ipv4Addr, SocketAddrV4, TcpListener};
+use std::ops::Range;
+use std::sync::Mutex;
+
+static PORT_IDX: Lazy<Mutex<u16>> = Lazy::new(|| Mutex::new(1000));
```

Comment:
Have you considered that this might not be necessary? Or that perhaps you could use say... `AtomicU16`? 
https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU16.html

Reply:
Not yet.